### PR TITLE
chore: use CircleCI

### PR DESCRIPTION
**Pull request by frederickfogerty**
This PR replaces the Travis CI config with an equivalent CircleCI config

### CODE REVIEW

**Comment by frederickfogerty** [comment]
Current dependencies on/for this PR:
* main
  * **PR #111**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Review by sherwinski — COMMENTED** [review]
Same comment as the [ruby PR](https://github.com/imgix/imgix-rb/pull/113#pullrequestreview-840543627), could we get some context around the parameter key setting and run steps shown here?

File: Gemfile [hunk 1]
```diff
@@ -2,3 +2,5 @@ source 'https://rubygems.org'
 
 # Specify your gem's dependencies in imgix-rails.gemspec
 gemspec
+
+gem "rspec_junit_formatter", "~> 0.4.1"
```
Comment (sherwinski) [inline Gemfile:6]: Should this be a dev dependency in the gemspec file?

Comment (frederickfogerty) [reply Gemfile:6]: Sure - added it over there

**Comment by frederickfogerty** [comment]
@sherwinski made some updates here, let me know what you think

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,12 +1,129 @@
...
+        type: string
+      path:
+        default: ./vendor/bundle
+        description: >
+          Installation path. By default, it will run bundle with `--deployment` flag
+          and installs gems to the vendor/bundle directory.
+        type: string
+      with-cache:
+        default: true
+        description: Enable automatic caching of your gemfile dependencies for increased speed.
```
Comment (luqven) [inline .circleci/config.yml:58]: Dope 👏🏼

**Review by luqven — COMMENTED** [review]
Orbs are a little over my head, but LGTM.

**Review by sherwinski — APPROVED** [review]
LGTM 👍

**Pull request merged**